docs/Deployment.md

```diff
@@ -10,51 +10,30 @@ BTCPay is a non-custodial invoicing system which eliminates the involvement of a
 Developer deployments are not recommended for production environments and require the user to have technical knowledge related to the build.
```

Comment:
> Developer deployments
It's unclear what developer deployments are?

Reply:
```suggestion
Manual deployments and Hardware builds are not recommended for production environments and require the user to have technical knowledge related to the build.
```

This should make it more understandable, and more relating to the titles `Manual deployment` and `Hardware build` shown in the new table.
Thoughts ?

Reply:
Yes, that's clearer indeed. :)

Reply:
Then committing that :)